components/Common/Button/index.tsx

```diff
@@ -0,0 +1,32 @@
+import classNames from 'classnames';
+import type { FC, ButtonHTMLAttributes } from 'react';
+
+import styles from './index.module.scss';
+
+type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
+  variant?: 'primary' | 'secondary' | 'tertiary';
+  special?: boolean;
+};
+
+const Button: FC<ButtonProps> = ({
+  variant = 'primary',
+  special = false,
+  children,
+  className,
+  ...props
+}) => {
+  const buttonStyles = classNames(
```

Comment:
I feel we should have tailwind-merge here, to remove the conflicts between the className prop and the button's default styles.

Reply:
We have already discussed it here https://github.com/nodejs/nodejs.org/issues/5758

Reply:
I assume tailwind does the optimisations by itself? Since tailwind actually "compiles" its classnames during build/runtime; hence I'd imagine it already merging things?

But this is just a hypothesis

Reply:
No, it doesn't. We can try to use the [Important modifier](https://tailwindcss.com/docs/configuration#important-modifier) or [tailwind-merge](https://github.com/dcastil/tailwind-merge).

Reply:
@araujogui can you show us what that would look like in this component?

I am also reading through https://github.com/dcastil/tailwind-merge/blob/v1.14.0/docs/what-is-it-for.md etc in the meantime

Reply:
Interesting didn't know that `tailwind-merge` doesn't play nice with `@apply` modifier